sdk/core/core-amqp/src/util/typeGuards.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Helper TypeGuard that checks if something is defined or not.
+ * @param thing - Anything
+ * @internal
+ */
+export function isDefined<T>(thing: T | undefined | null): thing is T {
```

Comment:
should we replace the new `if (foo != null)` checks introduced in this PR with this helper instead?

Reply:
Updated to use isDefined instead of != null.